modules/client/examples/01-client/12-prepare-uninstallation.ts

```diff
@@ -0,0 +1,78 @@
+/* MARKDOWN
+---
+title: PrepareInstallation
+---
+
+### Prepare Uninstallation of a plugin
+
+*/
+
+import {
+  Client,
+  DaoDepositSteps,
+  DepositParams,
+  GasFeeEstimation,
+  PrepareUninstallationParams,
+  PrepareUninstallationSteps,
+  SetAllowanceSteps,
+  TokenType,
+} from "@aragon/sdk-client";
+import { context } from "../index";
+
+// Instantiate the general purpose client from the Aragon OSx SDK context.
+const client: Client = new Client(context);
+
+const prepareUninstallationParams: PrepareUninstallationParams = {
+  daoAddressOrEns: "0x1234567890123456789012345678901234567890", // my-dao.dao.eth
+  versionTag: {
+    build: 1,
+    release: 1,
+  },
+  pluginAddress: "0x2345678901234567890123456789012345678901",
+};
+
+// Estimate how much gas the transaction will cost.
+const estimatedGas: GasFeeEstimation = await client.estimation.prepareUninstallation(
+  prepareUninstallationParams,
+);
+console.log({ avg: estimatedGas.average, max: estimatedGas.max });
+
+// Deposit the ERC20 tokens.
+const steps = client.methods.prepareUninstallation(prepareUninstallationParams);
+for await (const step of steps) {
+  try {
+    switch (step.key) {
+      case PrepareUninstallationSteps.PREPARING:
+        console.log({ txHash: step.txHash });
+        break;
+      case PrepareUninstallationSteps.DONE:
+        console.log({
+          permissions: step.permissions,
+          pluginAddress: step.pluginAddress,
+          pluginRepo: step.pluginRepo,
+          versionTag: step.versionTag,
+        });
+        break;
+    }
+  } catch (err) {
+    console.error(err);
+  }
+}
+
+/* MARKDOWN
+Returns:
+```tsx
+{
+  txhash: "0xb1c14a49...3e8620b0f5832d61c"
+}
+{
+  permissions: ["0x123...", "0x123..."],
+  pluginAddress: "0x1234567890...",
+  pluginRepo: "0x1234567890..."
```

Comment:
Do we need to return these 2?

Reply:
We dont but they are part of the `applyUninstallation` encoder so I dont see a reason for not leaving them there